### ISSUE TITLE
New Rule: require-retry-error-logging

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#48]
## Research Evidence

From LLM code review patterns: LLMs often implement retry logic but forget to log why a retry happened, making debugging impossible.

## Description

When implementing retry logic, require logging of the error that triggered the retry before the retry attempt.

### Bad
```ts
async function fetchWithRetry(url: string, retries = 3) {
  for (let i = 0; i < retries; i++) {
    try {
      return await fetch(url);
    } catch (e) {
      if (i === retries - 1) throw e;
      continue; // Silent retry - no visibility into why
    }
  }
}
```

### Good
```ts
async function fetchWithRetry(url: string, retries = 3) {
  for (let i = 0; i < retries; i++) {
    try {
      return await fetch(url);
    } catch (e) {
      if (i === retries - 1) throw e;
      logger.warn('Fetch failed, retrying', { attempt: i + 1, url, error: e });
    }
  }
}
```

## Why This Matters for LLMs
- LLMs write retry logic frequently
- They often forget observability
- This catches a real production debugging gap

## Detection
- Detect loops containing try/catch with continue/retry
- Check that catch block has logging before retry

## Priority
Low-Medium - nice to have for observability
